enhancements/machine-api/aws_efa_networking.md

```diff
@@ -0,0 +1,253 @@
+
+As this feature is only intended to be used with specific high performance workloads, which should not be running on
+control plane hosts, we do not expect end users to require EFA network interfaces for their control plane machines.
+
+#### Users will have to configure their own security groups
+
+To allow the feature to work, the security groups that contain the Machines must be configured to allow all traffic
+between the hosts within the security group. This is not configured by default on an IPI cluster.
+Users must manually make this change via the AWS console/cli.
+This prerequisite should be documented as a manual step to enable the feature.
+
+Additionally, as we have future plans to manage security groups for IPI clusters, these manual changes to the security
+groups may be difficult to adopt into the future management system. However, there is nothing to stop users of IPI
+clusters making changes such as this already, so this kind of change must be considered for any adoption plans already.
+
+Additionally, IPI users are free to add additional security groups to their clusters post install and specify these
+additional groups within their Machine specs. We have no control over this today and again, this is something that must
+be planned for in the future security group management process.
+
+#### Using EFA requires additional software
+
+The EFA network interface types require special drivers to enable them to be used.
+Fortunately, the RHEL kernel already includes the relevant kernel modules to allow the adapter to be used transparently
+as a standard network adapter within user space. This means that no additional configuration is required within RHCOS
+to allow EFA instances to be used as regular OpenShift nodes, existing networking is unaffected.
+
+To take advantage of the EFA performance benefits, additional libfabric software must be deployed.
+This software is currently distributed by AWS and it will be up to the end user to deploy this software within their
+clusters. Red Hat will not initially support the libfabric driver.
+
+#### Only a subset of instance types are supported with EFA network interfaces
+
+Only a small subset of AWS instance types (eg. m5dn.24xlarge, m5dn.metal) are supported for use with EFA interfaces.
+If a user were to specify an EFA interface type with a non supported instance (eg. m5.xlarge), then the instance will
+fail to create. In this scenario, AWS rejects the RunInstance call from Machine API which causes Machine API to mark the
+Machine as `Failed`. An error message will be added to the Machine status to signal to the user that their configuration
+was invalid.
+
+> error launching instance: EFA interfaces are not supported on m5.2xlarge
+
+## Design Details
+
+### Open Questions
+
+- What support will we provide in the future?
+  - Will we be able to provide full end to end support for the required libfabric drivers
```

Comment:
```suggestion
  - Will we be able to provide full end to end support for the required libfabric drivers?
```

Reply:
I don't see how we can.  I would feel very uncomfortable biting off support for something we do not ship, and that we can not patch.

I think you need to make it clear that any issue encountered must be reproducible on a cluster without the libfabric daemonset present in order for us to support it.  If you want more than that... which team would be willing to take on the burden?